server/src/main/java/org/elasticsearch/action/index/IndexRequest.java

```diff
@@ -665,7 +683,7 @@ public String toString() {
                 sSource = "n/a, actual length: [" + new ByteSizeValue(source.length()).toString() + "], max length: " +
                     new ByteSizeValue(MAX_SOURCE_LENGTH_IN_TOSTRING).toString();
             } else {
-                sSource = XContentHelper.convertToJson(source, false);
+                sSource = XContentHelper.convertToJson(source, false, contentType);
```

Comment:
This is sort of a bug unrelated to this change but started tripping an assertion because we wouldn't always see a `BytesArray` here in tests now. The content recognition behaves slightly differently here when the source is a `BytesArray` and that caused a utf8 assertion to trip for `ReleasableBytesReference` now.